src/state.rs

```diff
@@ -95,6 +96,9 @@ pub struct ResolvedDnsStore {
     // in a future with support for per-pod DNS resolv.conf settings we may need
     // to change this to a map from source workload uid to resolved IP addresses.
     by_hostname: HashMap<String, ResolvedDns>,
+
+    // crd_hashmap is a map which store IP addresses for unkonwn domain after DNS resolving tasks
+    crd_hashmap: HashMap<String, SingleFlight>,
```

Comment:
nit: rename

btw, when will we cleanup the map

Reply:
okay, done!
Yeah, that's a good question, I think it should be sync with the hashmap: `by_hostname`.

Reply:
`crd_hashmap`  ---> `sf_by_hostname`